Modelica/Blocks/Logical.mo

```diff
@@ -929,6 +929,40 @@ The output <code>Q</code> is set by the input <code>S</code>, is reset by the in
 </html>"));
   end RSFlipFlop;
 
+  block AssertCondition "Assert that input u is true"
+    extends Modelica.Blocks.Interfaces.partialBooleanSI;
+    parameter String message = "Assert block failed"
+      "Assertion message to be displayed if the input u is false; please adapt";
+    parameter AssertionLevel assertionLevel=AssertionLevel.error
+      "Assertion level";
+  equation
+    assert(u,message,assertionLevel);
+    annotation (
+      Icon(graphics={
+          Text(
+            extent={{-90,40},{90,-40}},
+            textString="assert")}),
+      Documentation(
+        info="<html>
+<p>
+This block asserts that the Boolean input&nbsp;<strong>u</strong> is <strong>true</strong>.
+If&nbsp;<strong>u</strong> is <strong>false</strong>, the simulation will
+either abort with an error (if <strong>assertionLevel</strong> is set to
+<strong>error</strong>), or issue a&nbsp;warning and continue (if
+<strong>assertionLevel</strong> is set to <strong>warning</strong>).
+The <strong>message</strong> will be reported in either case.
+For further information, refer to the documentation of
```

Comment:
Just noting that the additional paragraph break suggested in https://github.com/modelica/ModelicaStandardLibrary/pull/3985#discussion_r953849225 is not here.  I don't have much of an opinion on whether it is a good idea; this is just an observation.

Reply:
The "further information" part was added to the first paragraph. The new paragraph starts after that.